ext/com_dotnet/com_handlers.c

```diff
@@ -353,7 +353,7 @@ static zend_function *com_method_get(zend_object **object_ptr, zend_string *name
 							ITypeComp_Release(bindptr.lptcomp);
 							break;
 
-						case DESCKIND_NONE:
+						default:
```

Comment:
Perhaps `EMPTY_SWITCH_DEFAULT_CASE()` instead?

Reply:
That macro is a midzomer and dangerous, it asserts the path is unreachable

Reply:
> it asserts the path is unreachable

Yes and this seems to be the desired behavior here. The documentation indicates that only the 4 `DESCKIND_*` cases that were previously handled may be returned. I'd rather crash if a new case appears instead of silently doing the wrong thing with it.

Reply:
The thing is, you won't necessarily crash. Anything can happen because taking an unreachable path is UB. This can either result in taking the wrong case or doing something completely absurd.

Reply:
Remember that assertions are turned into assumes in release builds.

Reply:
I'd be fine with an `assert()`, but [`__assume()`](https://learn.microsoft.com/en-us/cpp/intrinsics/assume) is dangerous, it's comparable to UB, if applied at the wrong place, and I wouldn't fully trust [that documentation](https://learn.microsoft.com/en-us/windows/win32/api/oaidl/nf-oaidl-itypecomp-bind). Maybe in this case it is better to explicitly list the alternatives, and add a `ZEND_ASSERT(0)` there.

Reply:
`ZEND_ASSERT(0)` essentially would do the same thing as marking it unreachable. I'd be okay with `assert(0)` but I predict someone's gonna turn that into `ZEND_ASSERT` on refactor in the future. `ZEND_ASSERT` is turned into an assume, and with 0 as an argument it's an assume on something that's false, the compiler can do anything. Can we please not do dangerous programming?

Reply:
D'oh! Of course, you're right.

Maybe we should actually change `ZEND_ASSERT()` to not be `ZEND_ASSUME()` in non-debug builds, and use `ZEND_ASSUME()` directly, where we are absolutely sure that the code is unreachable (in which case we can likely drop such code, and maybe address compiler warnings by other means). Or something like that.

Reply:
I did not know the difference between assume and assert, for me the assertions are there to ensure in debug builds that the preconditions, or the current state is what you expect it to be.

Reply:
Transforming assertions into assumes was done on purpose during the phpng development: https://github.com/php/php-src/commit/e1ab2cfe1c4c53798e6fdfd49320650dac240945
Changing that would have a performance impact as people specifically use it in that way (debug time assertions, production time optimizations)